### PR TITLE
new changes by hasanein most of the setting up table has been done:WIP

### PR DESCRIPTION
**Pull request by hasaninqh**
one component which is the setting up table component is under progrees

### CODE REVIEW

File: src/components/TableSetup.js [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
+    //let n = this.state.currentStep;
+    this.setState({
+      currentStep:index
+ 
+   })
+        
+   }
+  
+  render() {
+    console.log(data);
```
Comment (rajgthub) [inline src/components/TableSetup.js]: Remove the console.lo

File: src/components/TableSetup.js [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
+    const filteredItem = data.filter((item)=> item.id === this.state.currentStep)
+    console.log(filteredItem);
```
Comment (rajgthub) [inline src/components/TableSetup.js]: remove the line

File: src/components/TableSetup.js [hunk 3]
```diff
@@ -1,7 +1,41 @@
...
+class  TableSetup extends Component {
+  constructor(props){
+    super(props)
+    this.state={
+      currentStep:1
+    }
+  }
+   setCurrentStep=(index)=>{
+     console.log(index);
+    //let n = this.state.currentStep;
```
Comment (rajgthub) [inline src/components/TableSetup.js]: remove line 12 and line 13!

File: src/components/TableSettingPagination.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import data from '../data/tableSetting.json';
+const TableSettingPagination = (props)=>{
+    
+      return data.map((item,index) =>
+          <div>
+              <button onClick={()=>props.setCurrentStep(index)} value={item.id}>{item.id}</button>
```
Comment (rajgthub) [inline src/components/TableSettingPagination.js]: This index starts from 0 as it maps the array's index but we have index starting from 1 in the json file.    {
      "id": 1,
      "title": "Bendable chopping board",
      "url": "./table-images/table_1.png"
    
    }

Comment (rajgthub) [reply src/components/TableSettingPagination.js]: You need to add 1 to the index as 

> <div>
>       <button onClick={() => props.setCurrentStep(index + 1)} value={item.id}>
>         {item.id}
>       </button>
>     </div>

Comment (rajgthub) [reply src/components/TableSettingPagination.js]: or pass item.id

**Comment by Samatar26** [comment]
@hasaninqh Good job, like @rajgthub mentioned, you should remove the console.log's. And here are some other things to improve on/fix:

- The images are currently inside the root public folder, but you will probably want to have an assets folder inside the public folder as you'll be using more assets like images/fonts, etc.
- You seem to have generated a package-lock.json, whereas @rajgthub has generated a yarn-lock file, you will start running into issues when you use both npm and yarn, so it's best to stick to one package manager. 

Unfortunately I don't have a lot of time to review it properly as I'm at work lool, but if you put Raj's and these fixes in place, should be good 👍

**Pull request merged**